jsk_naoqi_robot/jsk_nao_startup/nodes/joy-client.l

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env roseus
+
+
+(ros::roseus "joy_client")
+(ros::load-ros-package "naoeus")
+
+
+(if (not (find-package "NAOQI_BRIDGE_MSGS"))
+    (make-package "NAOQI_BRIDGE_MSGS" :nicknames "NAOQI_MSGS"))
+
+(ros::advertise "/speech" std_msgs::String 1)
+(defun speak-jp (str &rest args)
+  (let ((msg (instance std_msgs::String :init)))
+    (send msg :data (apply #'format nil str args))
+    (ros::ros-info "speech ~A" (send msg :data))
+    (ros::publish "/speech" msg))
+  )
+
+(ros::advertise "/nao_tweet" std_msgs::String 1)
+(defun tweet (str &rest args)
+  (ros::publish "/nao_tweet" (instance std_msgs::String :init :data (apply #'format nil str args))))
+
+
+(ros::advertise "/nao_robot/pose/joint_angles" naoqi_bridge_msgs::JointAnglesWithSpeed 1)
+
+(setq *power-button-time* (ros::time-now))
+(setq *power-button-mode* "/nao_robot/pose/wakeup")
+(setq *power-button-state* 0)
+(ros::subscribe "/joy" sensor_msgs::Joy
+		#'(lambda (msg)
+		    (let* ((buttons (send msg :buttons))
+			   (axes (send msg :axes))
+			   (power-button-state (elt buttons 8))
+			   (rb-button-state (elt buttons 5))
+			   (b-button-state (elt buttons 1))
+			   (x-button-state (elt buttons 2))
+			   (y-button-state (elt buttons 3)))
+		      (print (list buttons power-button-state (ros::time- (ros::time-now) *power-button-time*)))
+		      (cond ((and (= *power-button-state* 1)
```

Comment:
Both `*power-button-state*` and `power-button-state` are required because we need to check the change of button value.
If only `power button status` is checked, servo on / off will be performed many times.